### PR TITLE
NAS-117108 / 22.12 / NAS-117108: Added custom form controls

### PR DESCRIPTION
**Pull request by RehanY147**
Make sure that on the apps page, when installing/editing an app, the fields that are meant to be hidden/were hidden on master and still hidden. There are some apps to test this on the team scale cloud vm.

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
Couldn't install `photoprism` or `home-assistant`. The save button stays disabled.

File: src/app/modules/ix-forms/components/ix-dynamic-form/classes/custom-untyped-form-field.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { CustomUntypedFormArray } from 'app/modules/ix-forms/components/ix-dynamic-form/classes/custom-untped-form-array';
+import { CustomUntypedFormControl } from 'app/modules/ix-forms/components/ix-dynamic-form/classes/custom-untped-form-control';
+import { CustomUntypedFormGroup } from 'app/modules/ix-forms/components/ix-dynamic-form/classes/custom-untyped-form-group';
+
+export type CustomUntypedFormField =
+    CustomUntypedFormArray |
```
Comment (undsoft) [inline src/app/modules/ix-forms/components/ix-dynamic-form/classes/custom-untyped-form-field.ts:6]: `|` usually does in the beginning of the line:
```
export type A =
  | B
  | C
```

**Review by undsoft — APPROVED** [review]
Didn't find any obvious regressions. Any tests we can add?

**Comment by RehanY147** [comment]
Not really. These changes don't impact how the form behaves. Just changes how code handles it.

**Pull request merged**